packages/nimble-components/src/table-column/date-text/types.ts

```diff
@@ -25,3 +25,10 @@ export type DayPeriodFormat = Intl.DateTimeFormatOptions['dayPeriod'];
 export type DateStyle = Intl.DateTimeFormatOptions['dateStyle'];
 export type TimeStyle = Intl.DateTimeFormatOptions['timeStyle'];
 export type HourCycleFormat = Intl.DateTimeFormatOptions['hourCycle'];
+
+export const Hour12Format = {
+    default: undefined,
+    twelveHour: 'twelve-hour',
```

Comment:
Since these are just string values we don't need the kebab-casing. Additionally, we might consider making the values even easier for a user to parse (e.g. '12hour', and '24hour').

Reply:
I agree that kebab-casing is not what we typically do, so I don't think that should be our choice here. I'm on the fence for _12_ vs _twelve_ though. We don't generally use numbers, but our API already is using `cutom-hour12`.

Reply:
I actually made the suggestion more to avoid the more verbose 'twenty-four`, where I think '24' is just _way_ easier to parse for a user.

Reply:
I hadn't seen Molly's response when I update the values to '12Hour` and '24Hour`. Now I'm noticing that that is not the capitalization you suggested. I was just using camel casing. Any objections to this?

FYI, the reason I used kebab-cased values in the first place was because I noticed our use of `AttributeHelpers.GetEnumNamesAsKebabCaseValues` in nimble-blazor. It's weird that we have (and use) that helper if our convention for values is not to use kebab casing.

Reply:
I believe I was incorrect in my statement about [kebab-casing](https://github.com/ni/nimble/blob/main/packages/nimble-components/CONTRIBUTING.md#attribute-naming-convention). I think I thought what I did because I came across some internal APIs that didn't use the kebab-casing, and moreover, there are very few cases where we have enum values that would require it. Given that, I think it would be appropriate for the values to be `12-hour` and `24-hour` (convention is for it to be lower-case as well).

Reply:
After discussion, we decided that we should be aligning as closely as possible with the JS API, meaning having a tri-state boolean rather than a new enum.